File: ru102j/redisolar/src/main/java/com/redislabs/university/dao/SiteDao.java
package com.redislabs.university.dao;

import com.redislabs.university.api.Coordinate;
import com.redislabs.university.api.Site;

import java.util.List;

public interface SiteDao {
    List<Site> findAll();
    List<Site> findByGeo(Coordinate coordinate, Double radius, String radiusUnit);
    Site findById(Long id);
    void insert(Site site);
}

File: ru102js/redisolar/daos/site_dao.js
const daoLoader = require('./daoloader');

const impl = daoLoader.loadDao('site');

const insert = site => impl.insert(site);

const findById = id => impl.findById(id);

const findAll = () => impl.findAll();

const findByGeo = query => impl.findByGeo(query);

module.exports = {
  insert,
  findById,
  findAll,
  findByGeo,
};
